Load docs fonts from Gradle CDN

eriwen opened the pull request:
This significantly improves time to first meaningful paint
according to Google's Lighthouse:
 - WOFF2 fonts are smaller than TTF
 - Fonts are loaded in parallel to the rest of CSS, instead of
   inline. Not only reducing the time to first paint, but also giving
   a chance for the browser to layout the DOM in parallel

This change basically stops our inlining TTF fonts and loads them from Google Fonts. I considered manually subsetting the fonts or loading with [webfontloader](https://github.com/typekit/webfontloader), but decided this a reasonable first step with performance benefits I am certain of.

### Context
I was digging into Google Analytics and found the load times of Gradle docs to be poor, especially in Asia (which is 26% of traffic)

![screenshot 2017-12-01 16 32 09](https://user-images.githubusercontent.com/51534/33517649-382ab578-d745-11e7-91cf-da9459fbf58f.png)

This results in 2 behavior changes when I compare them:
 * 30% fewer page views from Asia compared to all users
 * Many more visits (6x) to userguide_single.html compared to the rest of world, presumably so they don't have to load multiple pages to get most of the user guide content.

I'm suggesting this against `release` for Gradle 4.4 because if we can achieve even 10% more page views from Asia, that is 18,000 more views and I believe this is a low enough risk change that we can fix it separately from the Gradle distribution.

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

eriwen commented:
@ldaley Since you implemented the font stuff 5 years ago, I'm happy to have your review here if you're interested.

bmuschko commented on subprojects/docs/src/docs/stylesheets/dslHtml.xsl:
```diff
@@ -36,6 +36,7 @@
 
     <!-- customize the stylesheets to add to the <head> element -->
     <xsl:template name="output.html.stylesheets">
+        <link href="https://fonts.googleapis.com/css?family=Lato:400,700" rel="stylesheet"/>
```
Does this also work for offline documentation?

blindpirate replied:
BTW, China government blocks `googleapi` access...

eriwen replied:
@blindpirate I'd like to understand how other Gradle sites (or other sites using Google Fonts) behave for you. We use Google Fonts for the Blog and assets.gradle.com for other sites. 

Does the Gradle Blog load for you at all? If it does, do the requests to Google Fonts fail quickly, slowly, or succeed? Do our sites load slowly for you? How does this change when using versus not using the VPN?

bmuschko commented on subprojects/docs/docs.gradle:
```diff
@@ -186,16 +175,10 @@ tasks.withType(AssembleDslDocTask) {
 task configureCss {
     doLast {
-        def images = fileTree(dir: "src/docs/css/images", include: "*.*").files.collectEntries {
+        ext.tokens = fileTree(dir: "src/docs/css/images", include: "*.*").files.collectEntries {
```
I think the use of `ext` doesn't make too much sense here. I don't think access the specific action of the task from somewhere else to get ahold of field. In this case it's probably fine to have a free-standing variable. Do we reference `tokens` from somewhere else?

bmuschko commented:
> I'm suggesting this against release for Gradle 4.4 because if we can achieve even 10% more page views from Asia, that is 18,000 more views and I believe this is a low enough risk change that we can fix it separately from the Gradle distribution.

Can you clarify what you mean by "fix it separately from the Gradle distribution"? Do you mean if it's broken, then we can fix it separately? I am -1 on adding it to `release` this late.

eriwen commented:
I've rebased this on `master` for Gradle 4.5. I switched to the font weight @aweida had in the design spec, too. 🔥  This doesn't really do it justice.

![screenshot 2017-12-04 16 48 11](https://user-images.githubusercontent.com/51534/33582555-02ba4fe6-d913-11e7-9564-7a2ad1d9f932.png)

ldaley commented:
@eriwen back then, offline viewing was a (never used) hard requirement.

FWIW, for the gradle.com sites we are using https://assets.gradle.com/lato/css/lato-font.css and https://assets.gradle.com/inconsolata/inconsolata.css wherever practical. This is behind the Cloudflare CDN.

eriwen commented:
Thanks @ldaley. I've pulled in a subset of https://assets.gradle.com/lato/css/lato-font.css here (and renamed the pull request). This seems to have the added benefit of actually working in China (as opposed to Google Fonts) and I think the new Lato font is slick.

eriwen commented:
@bmuschko Please let me know what you think after I made a couple changes.

bmuschko commented:
@eriwen Looks more crisp and uses up less space. LGTM! Didn't see a response on my question about offline browsing. What does the behavior look like e.g. for the user guide bundled with the Gradle distribution?

eriwen commented:
@bmuschko PDF users will no longer get the embedded font, and it'll default to Helvetica or Arial. I don't consider this significant enough to address right now.

bmuschko commented:
@eriwen OK, thanks for clarifying. That's good enough.
